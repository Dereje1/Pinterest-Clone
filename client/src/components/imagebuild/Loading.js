import React from 'react';
import PropTypes from 'prop-types';

const Loading = ({ imagesLoaded, ready }) => (
  <div id="bubblecontainer" style={{ display: imagesLoaded && ready ? 'none' : 'flex' }}>
    <div className="bubbles A" />
    <div className="bubbles B" />
    <div className="bubbles C" />
  </div>
);

export default Loading;

Loading.defaultProps = {
  imagesLoaded: undefined,
  ready: undefined,
};

Loading.propTypes = {
  imagesLoaded: PropTypes.bool,
  ready: PropTypes.bool,
};
