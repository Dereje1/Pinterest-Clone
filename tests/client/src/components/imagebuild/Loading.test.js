import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Loading from '../../../../../client/src/components/imagebuild/Loading';

test('Will render the loading bubbles', () => {
  const wrapper = shallow(<Loading />);
  const bubbles = wrapper.find({ id: 'loadingcontainer' });
  expect(toJson(bubbles)).toMatchSnapshot();
});

test('Will not render the loading bubbles', () => {
  const wrapper = shallow(<Loading ready imagesLoaded />);
  const bubbles = wrapper.find({ id: 'loadingcontainer' });
  expect(toJson(bubbles)).toMatchSnapshot();
});
