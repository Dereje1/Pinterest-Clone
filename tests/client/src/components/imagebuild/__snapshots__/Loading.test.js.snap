// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Will not render the loading bubbles 1`] = `
<div
  id="loadingcontainer"
>
  <Memo(ForwardRef(DownloadDoneIcon)) />
</div>
`;

exports[`Will render the loading bubbles 1`] = `
<div
  id="loadingcontainer"
>
  <div
    className="bubbles A"
  />
  <div
    className="bubbles B"
  />
  <div
    className="bubbles C"
  />
</div>
`;
